assets/javascripts/modules/listCollapse.js

```diff
@@ -0,0 +1,61 @@
+/**
+  ListCollapse
+
+  Expand/collapse list content on smaller breakpoints, remain inline non interactive list at larger breakpoints.
+
+  Mandatory JS hooks:
+   * `data-list--collapse-toggle` - list toggle element hook to init ListCollapse click listener
+
+  Mandatory classes:
+   * `list--collapse` - placed on the outer container
+   * `list__item` - placed on the list items (eg li) in the list.
+
+  Optional Classes:
+   * `hide-inline-list--collapse-targets` - placed on the outer container to hide collapse the list by default. This class gets toggled when the toggle click listener fires
```

Comment:
I can't use data attributes (either data-list-collapse="open" or the aria-expanded approach) as I would need to be able to set the attribute property in css to retain the correct behaviour across breakpoints which I can't do. OR detect the breakpoint changes in js (blueegghhh :( ) which even then wouldn't work in non js browsers. Toggling a class in js which (in css) only has an effect at smaller breakpoints (hides and shows the expandable content) seems to be the most rubust approach across breakpoints and js/non js users.